diffsims/generators/diffraction_generator.py

```diff
@@ -52,27 +164,69 @@ class DiffractionGenerator(object):
     ----------
     accelerating_voltage : float
         The accelerating voltage of the microscope in kV.
-    max_excitation_error : float
-        Removed in this version, defaults to None
-    debye_waller_factors : dict of str:value pairs
-        Maps element names to their temperature-dependent Debye-Waller factors.
     scattering_params : str
         "lobato" or "xtables"
+    minimum_intensity : float
+        Minimum intensity for a peak to be considered visible in the pattern
+    precession_angle : float
+        Angle about which the beam is precessed. Default is no precession.
+    approximate_precession : boolean
+        When using precession, whether to precisely calculate average
+        excitation errors and intensities or use an approximation.    
+    shape_factor_model : function or string
+        A function that takes excitation_error and
+        `max_excitation_error` (and potentially kwargs) and returns
+        an intensity scaling factor. If None defaults to
+        `shape_factor_models.lorentzian`. A number of pre-programmed functions
+        are available via strings.
+    kwargs
+        Keyword arguments passed to `shape_factor_model`.
+
+    Notes
+    -----
+    * A full calculation is much slower and is not recommended for calculating
+      a diffraction library for precession diffraction patterns.
+    * When using precession and approximate_precession=True, the shape factor
+    model defaults to lorentzian; shape_factor_model is ignored. Only with
+    approximate_precession=False the custom shape_factor_model is used.
     """
 
     def __init__(
         self,
         accelerating_voltage,
-        max_excitation_error=None,
-        debye_waller_factors={},
         scattering_params="lobato",
+        precession_angle=None,
+        shape_factor_model=None,
```

Comment:
I think this is clearer if you write explicitly that it defaults to `"linear"`

Reply:
I'll leave it at linear, but since there is a reference to back up Lorenzian, I would vote to change this to be the default.